### ISSUE TITLE
TypedRecord constructors

### ISSUE THREAD

**Issue by AntonOresten** [anton083/TypedFASTX.jl#33]
The constructors are a complete mess at the moment (there are like 30 for TypedRecord), and a lot of it has to do with how the quality field of a TypedRecord can be either of type NoQuality or QualityScores. Consider making TypedRecord an abstract type, and differentiate between TypedRecords with and without qualities. The names of these types shouldn't be too long though, and they should be easy to create. There should still be type aliases like `DNARecord` such that it's as easy as possible to create records.

Here's an idea of how it might look:
```julia
using BioSequences

abstract type TypedRecord{T, Q <: AbstractQuality} end

struct TypedFASTARecord{T} <: TypedRecord{T, NoQuality}
    description::String
    sequence::T
end

struct TypedFASTQRecord{T} <: TypedRecord{T, QualityScores}
    description::String
    sequence::T
    quality::QualityScores
end

const DNARecord = TypedRecord{LongDNA{4}}
```
The constructors might end up looking just as ugly though, and readers and writers would require an overhaul as well.

Should there be constructors where you don't need to specify name/description of records?

**Comment by anton083** [comment]
I've implemented the proposed changes on [this branch](https://github.com/anton083/TypedFASTX.jl/tree/abstract-typedrecord)

**Comment by AntonOresten** [comment]
This is being implemented in v1.0.0. See PR #36.
